### DISCUSSION TITLE
Rich.number to show negative numbers in red and positive in green (themeable)

### DISCUSSION THREAD

**Discussion by pannet1** [Textualize/rich#3157] [Ideas] [answered]
Thanks for the awesome Rich TUI. Is it possible to provide number value aware Rich.number that can show negative values in red. This inturn could be tied up with the theme or be standalone.

**Comment by willmcgugan** [comment, marked as answer]
That's not something I would want to add to the core library, but it would be straightforward to implement. See the [Console Protocol](https://rich.readthedocs.io/en/latest/protocol.html) for details.

**Reply by pannet1** [reply]
ended up like this for trivial usage

  ```
class Regative:
    def __init__(self, val):
        self.val = val
        
    def __rich__(self) -> str:
        if self.val < 0:
            return f"[bold red]{self.val}"
        else:
            return f"[bold green]{self.val}"

 # usage
 pretty = f"{62* ' '}"
 print(pretty, "TOTAL:", Regative(totl), "\n")

```
